compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -144,6 +148,19 @@ public sealed class ClassReference {
       }
     }
 
+    override fun boundTypeOrNull(annotationFqName: FqName): FqName? =
+      clazz.boundTypeOrNull(annotationFqName, module)
+
+    override fun directSuperClassReferences(): Sequence<ClassReference> {
```

Comment:
Similar to `enclosingClassesWithSelf` I think we should simply cache a list and only query that information once.